### PR TITLE
rustc: rename ty::maps to ty::query.

### PR DESCRIPTION
**Pull request by eddyb**
For https://github.com/rust-lang/rust/pull/51550.

### CODE REVIEW

File: src/high-level-overview.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ input, running the type-checker, and so forth. This on-demand model
 permits us to do exciting things like only do the minimal amount of
 work needed to type-check a single function. It also helps with
 incremental compilation. (For details on defining queries, check out
-`src/librustc/ty/maps/README.md`.)
+`src/librustc/ty/query/README.md`.)
```
Comment (mark-i-m) [inline src/high-level-overview.md]: Actually could you change this to a link to the queries/incr comp chapter?

File: src/query.md [hunk 2]
```diff
@@ -215,14 +215,14 @@ Well, defining a query takes place in two steps:
 
 To specify the query name and arguments, you simply add an entry to
 the big macro invocation in
-[`src/librustc/ty/maps/mod.rs`][maps-mod]. This will probably have
+[`src/librustc/ty/query/mod.rs`][query-mod]. This will probably have
 changed by the time you read this README, but at present it looks
 something like:
```
Comment (mark-i-m) [inline src/query.md]: Could you please remove the last sentence (even though you didn't add it)?

**Comment by mark-i-m** [comment]
Ping @eddyb

**Comment by eddyb** [comment]
Somehow I didn't get notified on this, weird! (or I dismissed it thinking it was the rust-lang/rust PR)

**Comment by eddyb** [comment]
@mark-i-m Updated.

**Comment by mark-i-m** [comment]
Thanks! I will take a look tomorrow.

**Pull request merged**